config/config_helpers.go

```diff
@@ -447,15 +449,8 @@ func FindInParentFolders(
 	previousDir = filepath.ToSlash(previousDir)
 
 	if fileToFindParam == "" || fileToFindParam == DefaultTerragruntConfigPath {
-		if control, ok := strict.GetStrictControl(strict.RootTerragruntHCL); ok {
-			warn, triggered, err := control.Evaluate(ctx.TerragruntOptions)
-			if err != nil {
-				return "", err
-			}
-
-			if !triggered {
-				ctx.TerragruntOptions.Logger.Warnf(warn)
-			}
+		if err := ctx.TerragruntOptions.StrictControls.FilterByNames(controls.RootTerragruntHCL).Evaluate(log.ContextWithLogger(ctx, ctx.TerragruntOptions.Logger)); err != nil {
```

Comment:
This is really hard to read. We should split this over multiple lines so that it's easier to understand what's going on.

e.g.

```go
controls := ctx.TerragruntOptions.StrictControls
rootTGHCLControl := controls.FilterByNames(controls.RootTerragruntHCL)
logger := log.ContextWithLogger(ctx, ctx.TerragruntOptions.Logger)

if err := rootTGHCLControl.Evaluate(logger); err != nil {
...
}
```